### PR TITLE
C28120, C28121, and C28135

### PR DESCRIPTION
**Pull request by finchnest**
C28120, C28121, and C28135

** Most of the changed files will disappear once the previous PR gets merged.

### CODE REVIEW

File: src/drivers/wdm/experimental/IrqTooHigh/IrqTooHigh.ql [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/**
+ * @name IrqTooHigh
+ * @kind problem
+ * @description The function is not permitted to be called at the current IRQ level. The current level is too high.
+ * @problem.severity error
+ * @id cpp/portedqueries/irq-too-high
+ * @platform Desktop
+ * @repor.text The following line(s) potentially contains a function call that is supposed to be called at a higher Irql level.
```
Comment (NateD-MSFT) [inline src/drivers/wdm/experimental/IrqTooHigh/IrqTooHigh.ql:10]:
```suggestion
 * @repro.text The following line(s) potentially contains a function call that is supposed to be called at a higher Irql level.

```

File: src/drivers/PortFolder/build_create_analyze_test.cmd [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 call :test ExaminedValue WDMTestingTemplate wdm
 call :test StrSafe KMDFTestTemplate kmdf
-
-
+call :test MultiplePagedCode WDMTestingTemplate wdm
+call :test NoPagedCode WDMTestingTemplate wdm
+call :test NoPagingSegment WDMTestingTemplate wdm
 call :test OpaqueMdlUse WDMTestingTemplate wdm
 call :test OpaqueMdlWrite WDMTestingTemplate wdm
+call :test KeWaitLocal WDMTestingTemplate wdm
```
[the remark this answers is not in the document]
Comment (NateD-MSFT) [reply src/drivers/PortFolder/build_create_analyze_test.cmd:12]: Abenezer and I spoke again and a later commit resolved this.  Never mind!

**Review by lauragonzalezzz — APPROVED** [review]
Agree with Nate's points, nothing to add.

File: src/drivers/wdm/queries/KeWaitLocal/KeWaitLocal.ql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+import cpp
+
+from FunctionCall call, VariableAccess va
+where
+  call.getTarget().getName() = "KeWaitForSingleObject" and
+  call.getArgument(2).getValue().toInt() != 0 and
+  call.getArgument(0).(AddressOfExpr).getOperand() = va and
+  va.getTarget() instanceof LocalVariable
```
[the remark this answers is not in the document]
Comment (finchnest) [reply src/drivers/wdm/queries/KeWaitLocal/KeWaitLocal.ql:22]: @NateD-MSFT  I think this fix might address your question. What do you think?

`
/**
 * A C/C++ variable with _automatic storage duration_. In other words, a
 * function parameter or a local variable that is not static or thread-local.
 * For example, the variables `a` and `b` in the following code.
 * ```
 * void myFunction(int a) {
 *   int b;
 *   static int c;
 * }
 * ```
 */
class StackVariable extends LocalScopeVariable {
  StackVariable() {
    not this.isStatic() and
    not this.isThreadLocal()
  }
}
`

[the remark this answers is not in the document]
Comment (NateD-MSFT) [reply src/drivers/wdm/queries/KeWaitLocal/KeWaitLocal.ql]: The change you provided seems reasonable to me.

**Review by NateD-MSFT — APPROVED** [review]

**Pull request merged**